Pull request for issue #87

luizbon opened the pull request:
Just modified the Create-Release script to find nuget.config that determines where packages are.

luizbon committed:
Added support to find nuget.config file and use it to find the packages folder

shiftkey commented on src/CreateReleasePackage/tools/Create-Release.ps1:
```diff
@@ -63,7 +63,7 @@ function Create-ReleaseForProject {
         echo "Found package $pkgFullName"
 
-		$packageDir = Join-Path $solutionDir "packages"
+		$packageDir = Get-Nuget-Packages-Path(Join-Path $solutionDir "packages")
```
Should this be passing in _just_ the solution directory?

shiftkey commented on src/CreateReleasePackage/tools/Create-Release.ps1:
```diff
@@ -88,6 +88,28 @@ function Create-ReleaseForProject {
 	}
 }
 
+function Get-Nuget-Packages-Path {
```
PowerShell functions are generally named as `Verb-TheRest` - let's make this `Get-NuGetPackagesPath` for extra clarity

shiftkey commented on src/CreateReleasePackage/tools/Create-Release.ps1:
```diff
@@ -88,6 +88,28 @@ function Create-ReleaseForProject {
+        return $path.value
+    }
+
+    $parent = Split-Path $solutionDir
```
Do we need to loop and recurse here? I think we'll only ever have one file to read, so why not:
- if we have a config, find it's `repositoryPath` value. 
- does it exist? use it.
- otherwise return `Join-Path $solutionDir "packages"`

luizbon committed:
PowerShell functions are named as Verb-TheRest

PowerShell functions are generally named as Verb-TheRest - let's make
this Get-NuGetPackagesPath for extra clarity

shiftkey commented on src/CreateReleasePackage/tools/Create-Release.ps1:
```diff
@@ -63,7 +63,7 @@ function Create-ReleaseForProject {
         echo "Found package $pkgFullName"
 
-		$packageDir = Join-Path $solutionDir "packages"
+		$packageDir = Get-NugetPackagesPath(Join-Path $solutionDir "packages")
```
I think the `Join-Path` should be dropped and moved into the actual method, otherwise it's going to start searching inside `packages`.

The docs seem to suggest that the `nuget.config` file should be at same level as the solution.

luizbon committed:
Bug when don't exists nuget.config

If don't exists nuget.config return empty and use solutionDir + packages
as default directory

shiftkey commented on src/CreateReleasePackage/tools/Create-Release.ps1:
```diff
@@ -88,6 +91,28 @@ function Create-ReleaseForProject {
+        [string]$solutionDir
+    )
+
+    $cfg = ls $solutionDir -ErrorAction SilentlyContinue | ?{ $_.Name -eq "nuget.config" } | Select-Object -first 1
```
Even simpler: `$cfg = Get-ChildItem -Path $solutionDir -Filter nuget.config | Select-Object -first 1`

When it can't be found `$cfg` is null

EDIT: $solutionDir is what you need here. My bad.

shiftkey commented on src/CreateReleasePackage/tools/Create-Release.ps1:
```diff
@@ -63,7 +63,10 @@ function Create-ReleaseForProject {
 
-		$packageDir = Join-Path $solutionDir "packages"
+		$packageDir = Get-NugetPackagesPath($solutionDir)
+        if(-not $packageDir) {
```
Spaces instead of tabs? (line up this with the previous line)

luizbon committed:
minor ajusts for simplicity

luizbon committed:
Test if nuget.config without repositorypath

shiftkey commented:
Tidied up and additional testing over here: #107
